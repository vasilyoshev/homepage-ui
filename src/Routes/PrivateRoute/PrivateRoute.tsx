import { Route, Redirect, RouteProps } from 'react-router-dom';

export const PrivateRoute: React.FC<RouteProps> = ({ ...rest }) => {
  const isAuthed = !!localStorage.getItem('token');
  if (!isAuthed) {
    return <Redirect to='/login' />;
  }
  return <Route {...rest } />;
};
