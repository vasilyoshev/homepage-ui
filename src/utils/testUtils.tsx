import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

// TODO extract mocks to separate files, e.g. widgetsMock.js
const mockState = {
  loading: {},
  widgets: { widgetsProps: [] },
  user: {},
};
const createMockStore = configureStore([thunk]);
const store = createMockStore(mockState);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const Providers = ({ children }: any) => (
  <Provider store={store}>
    <MemoryRouter initialEntries={['/']}>{children}</MemoryRouter>
  </Provider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

export { customRender as render };
