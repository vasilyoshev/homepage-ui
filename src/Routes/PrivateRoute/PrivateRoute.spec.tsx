import { render } from 'utils';
import { PrivateRoute } from './PrivateRoute';

describe('PrivateRoute', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<PrivateRoute />);

    expect(asFragment()).toMatchSnapshot();
  });
});
