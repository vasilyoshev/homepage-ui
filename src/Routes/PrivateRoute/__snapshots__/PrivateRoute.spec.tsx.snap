// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute renders successfully 1`] = `<DocumentFragment />`;
